typescript/src/electrum.ts

```diff
@@ -30,6 +30,10 @@ export interface Credentials {
    * Protocol used by the Electrum server.
    */
   protocol: "tcp" | "tls" | "ssl" | "ws" | "wss"
+  /**
+   * Additional options used by the Electrum server.
+   */
+  options?: any
```

Comment:
I'm not sure about adding this property to the `Credentials`. Shouldn't it be a separate Client's property?

```ts
type Options = object

export class Client implements BitcoinClient {
  private credentials: Credentials
  private options?: Options

  constructor(credentials: Credentials, options?: Options) {
    this.credentials = credentials
    this.options = options
  }


  static fromUrl(url: string, options?: Options): Client {
    const credentials = this.parseElectrumCredentials(url)
    return new Client(credentials, options)
  }

```

CC @lukasz-zimnoch

Reply:
Yep, I think @nkuba is right. Those are two separate beings.

Reply:
@nkuba d2bef3175c872710f687be1339c8bbd119239163